keras_nlp/models/bert/bert_models.py

```diff
@@ -248,21 +249,22 @@ def __init__(
         self.cls_token_index = cls_token_index
 
     def get_config(self):
-        config = super().get_config()
-        config.update(
-            {
-                "vocabulary_size": self.vocabulary_size,
-                "hidden_dim": self.hidden_dim,
-                "intermediate_dim": self.intermediate_dim,
-                "num_layers": self.num_layers,
-                "num_heads": self.num_heads,
-                "max_sequence_length": self.max_sequence_length,
-                "num_segments": self.num_segments,
-                "dropout": self.dropout,
-                "cls_token_index": self.cls_token_index,
-            }
-        )
-        return config
+        return {
+            "vocabulary_size": self.vocabulary_size,
+            "hidden_dim": self.hidden_dim,
+            "intermediate_dim": self.intermediate_dim,
+            "num_layers": self.num_layers,
+            "num_heads": self.num_heads,
+            "max_sequence_length": self.max_sequence_length,
+            "num_segments": self.num_segments,
+            "dropout": self.dropout,
+            "name": self.name,
+            "trainable": self.trainable,
```

Comment:
Why is `trainable` part of the config? Presumably that is something dynamic to adjusted to the context? Not 100% sure about `name` either.

Reply:
`get_config` will be used during model saving to propagate those options. You want `name` and `trainable` in there (try this our on a dense layer, for example).

Normally, for a layer, if you chain to `super()` you get all of this stuff. But in our case, we don't want the big nested generic nested config that a functional model gives you by default, because that will give you back a generic functional object, with none of our `Bert` python properties and methods on it.

Ideally, we would fix this all in core Keras, so there was a better supported flow for a "functional subclass." Possibly this would include a way to chain to super before defining your functional layers in `init`, and a way you could simply call `super()` here in `get_config` and get the same behavior as a normal layer. But I don't think we want to gate this on a core Keras feature.

I think without core Keras changes, the ways to do this are to either:
 - Include `name` and `trainable` explicitly.
 - Explicitly call `keras.Layer.get_config(self)` at the top of this function.

Reply:
Ah got you. Yeah they are persisted.